projects/plugins/jetpack/extensions/blocks/google-calendar/edit.js

```diff
@@ -19,183 +18,163 @@ import { getBlockDefaultClassName } from '@wordpress/blocks';
-		const { editedEmbed } = this.state;
-		const embedString = editedEmbed.trim();
-		let attributes;
-
-		if ( IFRAME_REGEX.test( embedString ) ) {
-			attributes = extractAttributesFromIframe( embedString );
-		} else if ( SHAREABLE_REGEX.test( embedString ) ) {
-			attributes = { url: convertShareableUrl( embedString ) };
-		} else {
-			attributes = { url: embedString };
-		}
 
-		if ( ! URL_REGEX.test( attributes.url ) ) {
-			noticeOperations.removeAllNotices();
-			noticeOperations.createErrorNotice(
-				__(
-					"Your calendar couldn't be embedded. Please double check your URL or Embed Code. Please note, you need to use the 'Public URL' or 'Embed Code', the 'Shareable Link' will not work.",
-					'jetpack'
-				)
-			);
+		const newAttributes = parseEmbed( editedEmbed.trim() );
+
+		if ( ! URL_REGEX.test( newAttributes.url ) ) {
+			setErrorNotice();
 			return;
 		}
 
-		setAttributes( attributes );
-		this.setState( { editingUrl: false } );
+		setAttributes( newAttributes );
+		setEditingUrl( false );
 		noticeOperations.removeAllNotices();
 	};
 
-	getEditForm = ( className, editedEmbed ) => {
-		return (
-			<form onSubmit={ this.handleEmbed } className={ className }>
-				<textarea
-					type="text"
-					value={ editedEmbed }
-					className="components-placeholder__input"
-					aria-label={ __( 'Google Calendar URL or iframe', 'jetpack' ) }
-					placeholder={ __( 'Enter URL or iframe to embed here…', 'jetpack' ) }
-					onChange={ event => this.setState( { editedEmbed: event.target.value } ) }
-				/>
-				<Button isSecondary isLarge type="submit">
-					{ _x( 'Embed', 'button label', 'jetpack' ) }
-				</Button>
-			</form>
-		);
-	};
-	/**
-	 * Render a preview of the Google Calendar embed.
-	 *
-	 * @returns {object} The UI displayed when user edits this block.
-	 */
-	render() {
-		const { attributes, className, name, noticeUI } = this.props;
-		const defaultClassName = getBlockDefaultClassName( name );
-		const { url, height } = attributes;
-		const { editedEmbed, interactive, editingUrl } = this.state;
-
-		const iframeHeight = this.props.isMobile ? '300' : height;
-
-		const html = `<iframe src="${ url }" style="border:0" scrolling="no" frameborder="0" height="${ iframeHeight }"></iframe>`;
-
-		const permissionsLink = (
-			<ExternalLink href="https://en.support.wordpress.com/google-calendar/">
-				{ __( 'Enable Permissions for the calendar you want to share', 'jetpack' ) }
-			</ExternalLink>
-		);
+	const getEditForm = ( formClassName ) => (
+		<form onSubmit={ handleEmbed } className={ formClassName }>
+			<textarea
+				type="text"
+				value={ editedEmbed }
+				className="components-placeholder__input"
+				aria-label={ __( 'Google Calendar URL or iframe', 'jetpack' ) }
+				placeholder={ __( 'Enter URL or iframe to embed here…', 'jetpack' ) }
+				onChange={ event => setEditedEmbed( event.target.value ) }
+			/>
+			<Button isSecondary isLarge type="submit">
```

Comment:
Another `isLarge` attribute that we can remove.

Reply:
Done. Thanks.